### PR TITLE
Copy interval values to new objects in collapse function, add copy fu…

### PR DESCRIPTION
**Pull request by mayerm94**
…nction to interval and uncertainty. Changes made so that interval value would not be changed where it is referenced elsewhere. https://jira.mitre.org/browse/BONNIE-1628

Pull requests into cql-execution require the following.
Submitter and reviewer should ✔ when done.
For items that are not-applicable, mark "N/A" and ✔.

[CDS Connect](https://cds.ahrq.gov/cdsconnect) and [Bonnie](https://github.com/projecttacoma/bonnie) are the main users of this repository. 
It is strongly recommended to include a person from each of those projects as a reviewer.

**Submitter:**
- [x] This pull request describes why these changes were made
- [x] Code diff has been done and been reviewed (it does not contain: additional white space, not applicable code changes, debug statements, etc.)
- [x] Tests are included and test edge cases
- [x] Tests have been run locally and pass
- [ ] Code coverage has not gone down and all code touched or added is covered.
- [ ] All dependent libraries are appropriately updated or have a corresponding PR related to this change

**Reviewer:**

Name:
- [ ] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [ ] The tests appropriately test the new code, including edge cases
- [ ] You have tried to break the code

### CODE REVIEW

File: src/datatypes/interval.coffee [hunk 1]
```diff
@@ -13,6 +13,16 @@ module.exports.Interval = class Interval
     isInterval:
       get: -> true
 
+  copy: ->
+    newLow = @low
+    newHigh = @high
+    if @low.copy?()
```
Comment (cmoesel) [inline src/datatypes/interval.coffee]: In this `if` clause (and the one below), your if condition actually _invokes_ the `copy` function -- which means that you're creating a copy (and temporarily allocating memory) just to test if the copy function exists (and if it does, then you _invoke_ copy again when you assign it).

I'd recommend you just test to see if a copy function exists like this:

```coffee
if typeof @low.copy == 'function'
```

Comment (cmoesel) [reply src/datatypes/interval.coffee]: Same comment applies in several places throughout this PR.

**Comment by jbradl11** [comment]
Along with Moesel's comments, tests need to be added.

**Comment by cmoesel** [comment]
Odd thing.  When I pulled this down and ran `yarn` (which, in turn, runs the `build-cql4browsers` task), it generated a _different_ cql4browsers.js than is checked in on this PR.

@mayerm94 - can you double-check this on your end?  Try running the cql4browsers via `yarn` and see if you get a diff.  If so, can you push it?
